src/types.ts

```diff
@@ -1,12 +1,57 @@
+enum AddressType {
+  CANONICAL = 'canonical',
+  EIP155 = 'eip155',
+  ZKSYNC = 'zksync',
+}
+
+export interface SingletonDeploymentJSON {
+  released: boolean;
+  contractName: string;
+  version: string;
+  codeHash: string;
+  networkAddresses: Record<string, string | string[]>;
+  addresses: Record<string, string>;
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  abi: any[];
+}
+
+interface AbiInput {
+  indexed?: boolean;
+  internalType: string;
+  name: string;
+  type: string;
+}
+
+interface AbiItem {
```

Comment:
I would be careful with changing the types. In particular it might break packages as it might not be usable with `ethers` for example (I don't know if it uses slightly stricter tagged union types that discriminate on the `type` field) and would leave the change out of the PR TBH.

Reply:
I think you're right that this might be a breaking change. I reverted it for now, but I think we should include this in one of the future releases.